src/engine/InputQuery.cpp

```diff
@@ -255,18 +255,28 @@ InputQuery &InputQuery::operator=( const InputQuery &other )
                                   other._networkLevelReasoner->getConstraintsInTopologicalOrder().size() ).ascii() );
 
         unsigned numberOfDisjunctions = 0;
+        unsigned numberOfMaxs = 0;
         for ( const auto &constraint : other._plConstraints )
         {
             if ( constraint->getType() == DISJUNCTION )
-             {
-                 auto *newPlc = constraint->duplicateConstraint();
-                 _plConstraints.append( newPlc );
-                 ++numberOfDisjunctions;
-             }
+            {
+                auto *newPlc = constraint->duplicateConstraint();
+                _plConstraints.append( newPlc );
+                ++numberOfDisjunctions;
+            }
+            else if ( constraint->getType() == MAX &&
```

Comment:
I think we assume that network level reasoner can pick up all the Max constraint. Maybe just remove the logic for Max here?

Reply:
Your understanding is right.
However, when the network reasoner is true, the below part also picks up max constraints.
To avoid to pick up doubly, this neglects max constraints here.

```python
        for ( const auto &constraint : other._networkLevelReasoner->
                  getConstraintsInTopologicalOrder() )
        {
            auto *newPlc = constraint->duplicateConstraint();
            _plConstraints.append( newPlc );
            _networkLevelReasoner->addConstraintInTopologicalOrder( newPlc );
        }
```

Does it make sense?